source/daplink/drag-n-drop/vfs_user.c

```diff
@@ -318,28 +299,85 @@ void vfs_user_disconnecting()
 static uint32_t get_file_size(vfs_read_cb_t read_func)
 {
     // Determine size of the file by faking a read
-    return read_func(0, file_buffer, 1);
+    return read_func(0, NULL, 0);
 }
 
-// File callback to be used with vfs_add_file to return file contents
-static uint32_t read_file_mbed_htm(uint32_t sector_offset, uint8_t *data, uint32_t num_sectors)
-{
-    if (sector_offset != 0) {
-        return 0;
+#ifndef EXPANSION_BUFFER_SIZE
+#define EXPANSION_BUFFER_SIZE 128
+#endif
+
+uint32_t expand_string_in_region(uint8_t *buf, uint32_t size, uint32_t start, uint32_t pos, uint8_t *input) {
+    uint8_t str_buf[EXPANSION_BUFFER_SIZE];
+    memset(str_buf, 0, sizeof(str_buf));
+    for(uint32_t i = 0; (i < (sizeof(str_buf) - 1)) && 0 != input[i]; i++) {
+        str_buf[i] = input[i];
     }
+    uint32_t l = expand_info(str_buf, sizeof(str_buf));
+
+    return util_write_in_region(buf, size, start, pos, str_buf, l);
+}
+
+uint32_t string_field_in_region(uint8_t *buf, uint32_t size, uint32_t start, uint32_t pos, uint8_t *label, const uint8_t *value) {
+    uint32_t l = util_write_string_in_region(buf, size, start, pos, label);
+    l += util_write_in_region(buf, size, start, pos + l, ": ", 2);
+    l += util_write_string_in_region(buf, size, start, pos, value);
+    l += util_write_in_region(buf, size, start, pos + l, "\r\n", 2);
+    return l;
+}
+
+uint32_t setting_in_region(uint8_t *buf, uint32_t size, uint32_t start, uint32_t pos, uint8_t *label, uint32_t boolean) {
```

Comment:
Would it make sense to move some of these into the `util_*` file as they are helper functions?

Reply:
I thought about it, but I did not think they would be useful outside of `vfs_user.c` and I would have to add a `util_` prefix to function name already quite long. On the upside `vfs_user.c` is already quite long so it would be useful to remove things from there.